src/helpers.js

```diff
@@ -221,15 +221,55 @@ module.exports = function(options) {
           } with offset ${offset}`
         );
 
-        const width = properties.width - offset * 2;
-        const height = properties.height - offset * 2;
+        const width =
+          (properties.rotate ? properties.height : properties.width) -
+          offset * 2;
+        const height =
+          (properties.rotate ? properties.width : properties.height) -
+          offset * 2;
         const svgSource = sourceset.find(source => source.size.type === "svg");
 
         let promise = null;
 
         if (svgSource) {
+          const background = { r: 0, g: 0, b: 0, alpha: 0 };
+
           log("Image:render", `Rendering SVG to ${width}x${height}`);
-          promise = svg2png(svgSource.file, { height, width }).then(Jimp.read);
+          promise = sharp(svgSource.file)
+            .resize({
+              background,
+              width,
+              height,
+              fit: sharp.fit.inside
+            })
+            .toBuffer({ resolveWithObject: true })
+            .then(({ data, info }) => {
+              if (info.width < width) {
+                return sharp(data)
+                  .extend({
+                    background,
+                    top: 0,
+                    bottom: 0,
+                    left: (width - info.width) >> 1,
+                    right: (width - info.width + 1) >> 1
+                  })
+                  .toBuffer()
+                  .then(Jimp.read);
+              }
+              if (info.height < height) {
+                return sharp(data)
+                  .extend({
+                    background,
+                    left: 0,
+                    right: 0,
+                    top: (height - info.height) >> 1,
+                    bottom: (height - info.height + 1) >> 1
+                  })
+                  .toBuffer()
+                  .then(Jimp.read);
+              }
```

Comment:
please avoid all this code duplication by unconditionally extending the image and relying on `sharp` to properly handle zero offset:

```.js
.then(({ data, info }) => sharp(data)
  .extend({
    background,
    top: (height - info.height) >> 1,
    bottom: (height - info.height + 1) >> 1
    left: (width - info.width) >> 1,
    right: (width - info.width + 1) >> 1
  })
  .toBuffer()
  .then(Jimp.read)
```

Reply:
Fixed